src/common/types/types.cpp

```diff
@@ -416,7 +471,12 @@ LogicalType::LogicalType(LogicalTypeID typeID) : typeID{typeID}, extraTypeInfo{n
 
 LogicalType::LogicalType(LogicalTypeID typeID, std::unique_ptr<ExtraTypeInfo> extraTypeInfo)
     : typeID{typeID}, extraTypeInfo{std::move(extraTypeInfo)} {
-    physicalType = getPhysicalType(typeID);
+    if (typeID == LogicalTypeID::DECIMAL) {
+        // special case, since decimal physical type is determined by its extratypeinfo
+        physicalType = DecimalType::getPhysicalTypeID(this->extraTypeInfo);
```

Comment:
Then we should change the getPhysicalType interface to also take extraTypeInfo

Reply:
As a whole LogicalType or just (LogicalTypeID, extraTypeInfo)?